cel/program.go

```diff
@@ -213,7 +213,9 @@ func newProgram(e *Env, ast *Ast, opts []ProgramOption) (Program, error) {
 		factory := func(state interpreter.EvalState, costTracker *interpreter.CostTracker) (Program, error) {
 			costTracker.Estimator = p.callCostEstimator
 			costTracker.Limit = p.costLimit
-			decs := decorators
```

Comment:
can we avoid the initial alloc and copy with a three slice variant that ensures decs has cap equal to len so any future appends guarantee a reallocate?

`decs := decorators[:len(decorators):len(decorators)]`

Reply:
I can see how this bug didn't surface until we started supporting cost tracking since the exhaustive eval and state observation are idempotent (memory leaks though for eval state tracking). Good catch @inteon. 

@jpbetz @cici37 which versions of cel-go will you need this ported to?

Reply:
@liggitt I updated the code with your proposed fix.

Reply:
> @jpbetz @cici37 which versions of cel-go will you need this ported to?

Looks like:
* v0.10.x (for Kubernetes [1.24](https://github.com/kubernetes/kubernetes/blob/release-1.24/go.mod#L177))
* v0.12.x (for Kubernetes [1.25](https://github.com/kubernetes/kubernetes/blob/release-1.25/go.mod#L179) and [1.26](https://github.com/kubernetes/kubernetes/blob/release-1.26/go.mod#L175))